### PR TITLE
[fix] when adding new user external account, permission sync was not scheduled

### PR DESCRIPTION
**Pull request by kopancek**
We now schedule high priority permission sync when new external account is associated with the user.

## Test plan

Unit tests + manual test:
1. Add new user external account to existing user on Account Security page
2. Go to permissions
3. You should see a new high priority job with `Third-party login service added for the user` as a reason (which might be hidden in elipsis, see screenshot).
Screenshot 2023-06-26 at 13 16 38

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -102,6 +102,7 @@ All notable changes to Sourcegraph are documented in this file.
 - The braindot menu on the blob view no longer fetches data eagerly to prevent performance issues for larger monorepo users. [#53039](https://github.com/sourcegraph/sourcegraph/pull/53039)
 - Fixed an issue where commenting out redacted site-config secrets would re-add the secrets. [#53152](https://github.com/sourcegraph/sourcegraph/pull/53152)
 - Fixed an issue where SCIP packages would sometimes not be written to the database, breaking cross-repository jump to definition. [#53763](https://github.com/sourcegraph/sourcegraph/pull/53763)
+- When adding a new user external account, permission sync for the user was not scheduled. [#54144](https://github.com/sourcegraph/sourcegraph/pull/54144)
```
Comment (erzhtor) [inline CHANGELOG.md:105]:
```suggestion
- Fixed an issue when adding a new user external account was not scheduling a new permission sync for the user. [#54144](https://github.com/sourcegraph/sourcegraph/pull/54144)
```

**Review by erzhtor — APPROVED** [review]
👍

File: internal/extsvc/gerrit/externalaccount/externalaccount.go [hunk 2]
```diff
@@ -43,5 +47,13 @@ func AddGerritExternalAccount(ctx context.Context, db database.DB, userID int32,
...
 		return err
 	}
+
+	logger := log.Scoped("AddGerritExternalAccount", "Add Gerrit External Account to existing user")
+	// Schedule a permission sync, since this is a new account for the user
+	permssync.SchedulePermsSync(ctx, logger, db, protocol.PermsSyncRequest{
+		UserIDs:           []int32{userID},
+		Reason:            database.ReasonExternalAccountAdded,
+		TriggeredByUserID: userID,
+	})
```
Comment (pjlast) [inline internal/extsvc/gerrit/externalaccount/externalaccount.go:57]: Does this mean you need to remove this scheduler call over [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@a338bd5801c90ee8f64c092fff5d5f9bf673a5dc/-/blob/cmd/frontend/graphqlbackend/external_accounts.go?L184-187) ?

Comment (kopancek) [reply internal/extsvc/gerrit/externalaccount/externalaccount.go:57]: Hmm, no, I probably need to do it here :D Good catch, I wanted to check where this function is called and then forgot.

File: internal/database/external_accounts.go [hunk 3]
```diff
@@ -309,6 +310,8 @@ func (s *userExternalAccountsStore) CreateUserAndSave(ctx context.Context, newUs
 }
 
 func (s *userExternalAccountsStore) Insert(ctx context.Context, userID int32, spec extsvc.AccountSpec, data extsvc.AccountData) (err error) {
+	// permssync.SchedulePermsSync
+
```
Comment (pjlast) [inline internal/database/external_accounts.go:314]: Leftover comment?

Comment (kopancek) [reply internal/database/external_accounts.go:314]:
```suggestion
```

**Pull request merged**